src/data/automation.ts

```diff
@@ -14,17 +15,28 @@ export interface AutomationEntity extends HassEntityBase {
   };
 }
 
-export interface AutomationConfig {
+export type AutomationConfig =
+  | ManualAutomationConfig
+  | BlueprintAutomationConfig;
+
+export interface BaseAutomationConfig {
   id?: string;
-  alias: string;
-  description: string;
+  alias?: string;
+  description?: string;
+}
+
+export interface ManualAutomationConfig extends BaseAutomationConfig {
   trigger: Trigger[];
   condition?: Condition[];
   action: Action[];
   mode?: "single" | "restart" | "queued" | "parallel";
   max?: number;
 }
 
+export interface BlueprintAutomationConfig extends BaseAutomationConfig {
```

Comment:
Blueprint can contain anything of a manual automation config, it will override whatever the blueprint makes.

```suggestion
export interface BlueprintAutomationConfig extends ManualAutomationConfig {
```

Reply:
Means we can get rid of `BaseAutomationConfig`

Reply:
We don't have to support this in the UI though !